File: algorithms/sorting/bogosort/java/src/main/java/com/thebelava/algorepo/sort/Bogosort.java
package com.thebelava.algorepo.sort;


import java.util.Random;


public class Bogosort {
    private final Random random = new Random();

    public int[] sort(int[] inputData) {
        if (isNothingToSort(inputData)) {
            return inputData;
        }
        while (isNotSorted(inputData)) {
            shuffleRandomly(inputData);
        }
        return inputData;
    }

    private boolean isNotSorted(int[] inputData) {
        for (int i = 0; i < inputData.length - 1; i++) {
            if (inputData[i] > inputData[i + 1]) {
                return true;
            }
        }
        return false;
    }

    private void shuffleRandomly(int[] inputData) {
        for (int i = 0; i < inputData.length; i++) {
            int currentValue = inputData[i];
            int randomIndex = random.nextInt(inputData.length);
            inputData[i] = inputData[randomIndex];
            inputData[randomIndex] = currentValue;
        }
    }

    private boolean isNothingToSort(int[] inputData) {
        return inputData == null || inputData.length <= 1;
    }

}

File: algorithms/sorting/bogosort/python/bogosort.py
import random


def bogosort(input_data):
    if len(input_data) <= 1:
        return input_data
    while is_not_sorted(input_data):
        shuffle_randomly(input_data)
    return input_data


def is_not_sorted(input_data):
    for i in range(0, len(input_data) - 1):
        if input_data[i] > input_data[i + 1]:
            return True
    return False


def shuffle_randomly(input_data):
    for i in range(0, len(input_data) - 1):
        random_index = random.randint(0, len(input_data) - 1)
        input_data[i], input_data[random_index] = input_data[random_index], input_data[i]
